File: KhumaloCraft/Entities/EntityClass.fs
namespace KhumaloCraft.Entities
//  entity class used with DbContext class

open System

type User() =
    member val UserID = 0 with get, set // Primary Key
    member val Email = "" with get, set // Unique
    member val PasswordHash = "" with get, set
    member val CreatedDate = DateTime.Now with get, set

type Product() =
    member val ProductID = 0 with get, set // Primary Key
    member val ProductName = "" with get, set
    member val Description = "" with get, set
    member val Price = 0.0 with get, set
    member val ImageUrl = "" with get, set

type ContactMessage() =
    member val MessageID = 0 with get, set // Primary Key
    member val Name = "" with get, set
    member val Email = "" with get, set
    member val Message = "" with get, set
    member val CreatedDate = DateTime.Now with get, set

File: Models/EntityClass.cs
using System.ComponentModel.DataAnnotations;

namespace KhumaloCraft.Models
{
    public class User
    {
        public int UserID { get; set; }

        [Required]
        [EmailAddress]
        public string UserEmail { get; set; }

        [Required]
        public string PasswordHash { get; set; }

        public DateTime CreatedDate { get; set; } = DateTime.Now;

        [Required]
        public string Role { get; set; } 
    }

    public class Product
    {
        public int ProductID { get; set; }
        public string ProductName { get; set; }
        public string Description { get; set; }
        public double Price { get; set; }
        public string ImageUrl { get; set; }
        public string Category { get; set; } 
        public int Stock { get; set; } 

        
    }

    public class ContactMessage
    {
        [Key] public int MessageID { get; set; }
        public string Name { get; set; }
        public string Email { get; set; }
        public string Message { get; set; }
        public DateTime CreatedDate { get; set; } = DateTime.Now;
    }

    public class PurchasedItem
    {
        public int ID { get; set; }
        public int ProductID { get; set; }
        public int Quantity { get; set; }

        // Navigation property to the Product entity
        public Product Product { get; set; }
    }
}
